### PR TITLE
Added offloading support FP8 attention

### PR DESCRIPTION
**Pull request by sanandaraj5597**
This PR allows offloading the QKV activation tensors for FP8 Attention.

### CODE REVIEW

**Review by timmoon10 — APPROVED** [review]
LGTM.

Can CPU offloading handle recieving `None` in the tensor list? https://github.com/NVIDIA/TransformerEngine/pull/1143 adds some cases where the FP8 tensors are not saved.

File: transformer_engine/pytorch/attention.py [hunk 1]
```diff
@@ -5637,16 +5637,27 @@ def forward(
...
+            if ctx.fp8:
+                tensor_list = []
+                for t in fp8_tensors:
+                    tensor_list.append(t)
+                for t in aux_ctx_tensors:
+                    tensor_list.append(t)
+            else:
+                tensor_list = [q, k, v, out_save]
+                for t in aux_ctx_tensors:
+                    tensor_list.append(t)
```
Comment (ksivaman) [inline transformer_engine/pytorch/attention.py:5654]:
```suggestion
            if ctx.fp8:
                tensor_list = fp8_tensors.extend(aux_ctx_tensors)
            else:
                tensor_list = [q, k, v, out_save].extend(aux_ctx_tensors)
```

Comment (ksivaman) [reply transformer_engine/pytorch/attention.py:5654]: Avoiding loops and make more compact

**Comment by sanandaraj5597** [comment]
> Can CPU offloading handle recieving `None` in the tensor list? #1143 adds some cases where the FP8 tensors are not saved.

Yes, it can handle.

**Review by ksivaman — APPROVED** [review]
Looks good

**Review by cyanguwa — APPROVED** [review]
LGTM

**Pull request merged**